### PR TITLE
AB-350: Update musicbrainz schema whenever actual MusicBrainz database is updated

### PR DESCRIPTION
**Pull request by rsh7**
This PR contains scripts to apply replications packets to musicbrainz schema in AcousticBrainz db to keep it updated with the actual Musicbrainz db.

### CODE REVIEW

File: db/import_mb_data.py [hunk 1]
```diff
@@ -9,6 +9,38 @@
...
+        """.format(table_name=table_name,
+                   columns=','.join(columns),
+                   value_str=join_columns(columns)))
+
+        result = connection.execute(query, values)
+        transaction.commit()
+
+
+def get_data_from_musicbrainz(table_name, data, column='id'):
+    with musicbrainz_db.engine.begin() as connection:
```
[the remark this answers is not in the document]
Comment (rsh7) [reply db/import_mb_data.py]: Actually, for the case of Integrity Error, we need to roll back a specific transaction and then start it again after getting the corresponding data.

File: db/import_mb_data.py [hunk 2]
```diff
@@ -9,6 +9,38 @@
...
+                 VALUES ({value_str})
+        """.format(table_name=table_name,
+                   columns=','.join(columns),
+                   value_str=join_columns(columns)))
+
+        result = connection.execute(query, values)
+        transaction.commit()
+
+
+def get_data_from_musicbrainz(table_name, data, column='id'):
```
[the remark this answers is not in the document]
Comment (rsh7) [reply db/import_mb_data.py]: Yeah, I intended to show you the code first and then work on docstrings.
In next commit :)

File: musicbrainz_importer/apply_replication_changes.py [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+                                    trans.rollback()
+                        if type == 'u':
+                            if keys or values:
+                                update_row(sql, params, connection, trans)
+                                print 'Updated rows in ' + table + ' table'
+                    print 'COMMIT; --', xid
+                # print ' - Statistics:'
+                # for table in sorted(stats.keys()):
+                #     print '   * %-30s\t%d\t%d' % (table, stats[table]['u'], stats[table]['d'])
+            print secsy
```
[the remark this answers is not in the document]
Comment (rsh7) [reply musicbrainz_importer/apply_replication_changes.py]: Done!

File: musicbrainz_importer/apply_replication_changes.py [hunk 4]
```diff
@@ -0,0 +1,264 @@
+import tarfile
```
[the remark this answers is not in the document]
Comment (rsh7) [reply musicbrainz_importer/apply_replication_changes.py]: :fireworks:

**Comment by rsh7** [comment]
@paramsingh, I am working on clean up for this script a bit. will push a commit tomorrow :)

**Comment by rsh7** [comment]
@brainzbot  retest this please

**Comment by rsh7** [comment]
@paramsingh It's displaying an operational error that 'acousticbrainz' doesn't exist. Can you please take a look? Any access grant problem might be there.

**Comment by paramsingh** [comment]
Yeah, that's probably me. I'll fix and rerun the tests

**Comment by rsh7** [comment]
Working now. :tada:

File: musicbrainz_importer/apply_replication_changes.py [hunk 5]
```diff
@@ -0,0 +1,405 @@
...
+    the database.
+
+    Args:
+        fileobj: tar file of the replication packet.
+        expected_schema_seq: The expected schema sequence that should be matched with the
+        one listed in replication packets.
+        replication_seq: The number of the replication packet.
+    """
+    print ("Processing", fileobj.name)
+    tar = tarfile.open(fileobj=fileobj, mode='r:bz2')
```
[the remark this answers is not in the document]
Comment (rsh7) [reply musicbrainz_importer/apply_replication_changes.py]: Closed the file.

File: db/import_mb_data.py [hunk 6]
```diff
@@ -8,13 +8,89 @@
...
+        data: data value whose corresponding row is fetched.
+        column: Column names for the tables. Take default as 'id' if not
+            specified.
+
+    Returns:
+        Table name, columns and data values fetched from the MusicBrainz
+        database.
+    """
+    with musicbrainz_db.engine.begin() as connection:
+        query = text("""
```
[the remark this answers is not in the document]
Comment (rsh7) [reply db/import_mb_data.py]: Okay, done! :)

File: musicbrainz_importer/apply_replication_changes.py [hunk 7]
```diff
@@ -0,0 +1,405 @@
...
+    """Store new replication sequence into replication_control table for future
+    updates and deletes from replication packets.
+
+    Args:
+        replication_seq: Current replication sequence to replace the old one.
+    """
+    with db.engine.begin() as connection:
+        query = text("""
+            UPDATE musicbrainz.replication_control
+               SET current_replication_sequence = %s""" % (replication_seq)
```
[the remark this answers is not in the document]
Comment (rsh7) [reply musicbrainz_importer/apply_replication_changes.py]: done!

**Comment by rsh7** [comment]
@paramsingh, How do I make table_name a parameter in [L55](https://github.com/metabrainz/acousticbrainz-server/pull/282/files#diff-cb389f8975027281a9c2a41ea44bdabeR55)?

**Pull request merged**